src/main/java/de/tum/cit/aet/artemis/core/service/user/UserService.java

```diff
@@ -325,6 +330,7 @@ public User registerUser(UserDTO userDTO, String password) {
             catch (VersionControlException e) {
                 log.error("An error occurred while registering GitLab user {}:", savedNonActivatedUser.getLogin(), e);
                 participationVCSAccessTokenService.deleteAllByUserId(savedNonActivatedUser.getId());
+                userSshPublicKeyService.deleteAllByUserId(savedNonActivatedUser.getId());
```

Comment:
Why is it necessary to do that? I don't think that new users will automatically have tokens and ssh keys?

Reply:
I agree, this is not necessary here. Neither is the call to `participationVCSAccessTokenService.deleteAllByUserId`, so I deleted it too.